helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -0,0 +1,253 @@
+  }
+
+  /**
+   * Set the cloudID field.
+   * @param cloudID
+   */
+  public void setCloudID(String cloudID) {
+    _record.setSimpleField(CloudConfigProperty.CLOUD_ID.name(), cloudID);
+  }
+
+  /**
+   * Get the CloudID field.
+   * @return CloudID
+   */
+  public String getCloudID() {
+    return _record.getSimpleField(CloudConfigProperty.CLOUD_ID.name());
+  }
+
+  /**
+   * Set the CLOUD_INFO_URLS field.
+   * @param cloudInfoURLs
+   */
+  public void setCloudInfoURLs(List<String> cloudInfoURLs) {
+    _record.setListField(CloudConfigProperty.CLOUD_INFO_URLS.name(), cloudInfoURLs);
+  }
+
+  /**
+   * Get the CLOUD_INFO_URLS field.
+   * @return CLOUD_INFO_URLS field.
+   */
+  public List<String> getCloudInfoURLs() {
+    return _record.getListField(CloudConfigProperty.CLOUD_INFO_URLS.name());
+  }
+
+  /**
+   * Set the CLOUD_INFO_PARSE_CLASS_NAME field.
+   * @param cloudInfoParserClassName
+   */
+  public void setCloudInfoParserClassName(String cloudInfoParserClassName) {
+    _record.setSimpleField(CloudConfigProperty.CLOUD_INFO_PARSE_CLASS_NAME.name(),
+        cloudInfoParserClassName);
+  }
+
+  /**
+   * Get the CLOUD_INFO_PARSE_CLASS_NAME field.
+   * @return CLOUD_INFO_PARSE_CLASS_NAME field.
+   */
+  public String getCloudInfoParserClassName() {
+    return _record.getSimpleField(CloudConfigProperty.CLOUD_INFO_PARSE_CLASS_NAME.name());
+  }
+
+  public static class Builder {
+    private String _clusterName = null;
+    private boolean _cloudEnabled = DEFAULT_CLOUD_ENABLED;
+    private String _cloudID;
+    private List<String> _cloudInfoURLList;
+    private String _cloudInfoParserClassName;
+
+    public CloudConfig build() {
+      validate();
+      return new CloudConfig(_clusterName, _cloudEnabled, _cloudID, _cloudInfoURLList, _cloudInfoParserClassName);
+    }
+
+    /**
+     * Default constructor
+     */
+    public Builder() {
+    }
+
+    /**
+     * Constructor with Cluster Name as input
+     * @param clusterName
+     */
+    public Builder(String clusterName) {
+      _clusterName = clusterName;
+    }
+
+    /**
+     * Constructor with CloudConfig as input
+     * @param cloudConfig
+     */
+    public Builder(CloudConfig cloudConfig) {
+      _cloudEnabled = cloudConfig.isCloudEnabled();
+      _cloudID = cloudConfig.getCloudID();
+      _cloudInfoURLList = cloudConfig.getCloudInfoURLs();
+      _cloudInfoParserClassName = cloudConfig.getCloudInfoParserClassName();
+    }
+
+
+    public Builder setClusterName (String v) {
+      _clusterName = v;
+      return this;
+    }
+
+    public Builder setCloudEnabled (boolean isEnabled) {
+      _cloudEnabled = isEnabled;
+      return this;
+    }
+
+    public Builder setCloudID (String v) {
+      _cloudID = v;
+      return this;
+    }
+
+    public Builder setCloudInfoURL (List<String> v) {
+      _cloudInfoURLList = v;
+      return this;
+    }
+
+    public Builder setCloudInfoURLField (String v) {
```

Comment:
If you prefer to provide function to put one URL at a time, let's call it addCloudInfoURL.

Reply:
Done.